fiovb/ta/entry.c

```diff
@@ -160,6 +162,89 @@ static TEE_Result delete_value(char *name, size_t name_sz)
 	return res;
 }
 
+static bool is_rollback_protected(void)
+{
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t value_sz = MAX_SIMPLE_VALUE_SIZE;
+	char value[MAX_SIMPLE_VALUE_SIZE];
+	uint32_t count;
+
+	res = read_value(ROLLBACK_PROT, strlen(ROLLBACK_PROT) + 1,
+			value, value_sz, &count);
+	if (res == TEE_SUCCESS) {
+		IMSG("Found %s, rollback protection is enabled",
+		      ROLLBACK_PROT);
+		return true;
+
+	} else if (res == TEE_ERROR_ITEM_NOT_FOUND) {
+		IMSG("Not found %s, rollback protection OTP is not blown",
+		      ROLLBACK_PROT);
+		return false;
+	} else {
```

Comment:
else not needed since we return inmediately....we could replade all else/else if with just if

Reply:
I would even use switch there.

Reply:
> I would like to distinguish two cases, when object is just not found or something wrong happened when reading the object value, and provide two different error messages

sure but you dont need the identation with the else since you are returning right above ... just a personal preference for readability in my case (else is just redundant IMO)